pep-9999.rst

```diff
@@ -0,0 +1,1037 @@
+     we don't propose a way to override. However, since the
+     base image is generally minimal, there shouldn't be
+     much of a use case for simply uninstalling packages
+     (especially without using the distribution's own
+     tools). The common case is when pip wants to upgrade a
+     package, which previously would have deleted the old
+     version (except on Debian). After this change, the old
+     version will still be on disk, but pip will
+     still _shadow_ externally-installed packages, and we
+     believe this to be sufficient for this not to be a
+     breaking change in practice - a Python ``import``
+     statement will still get you the newly-installed
+     package.
+
+     If it becomes necessary to have a way to do this, we
+     suggest that the distribution should document a way for
+     the installer tool to access the ``sysconfig`` scheme
+     used by the distribution itself.  See the
+     `Recommendations for distributions`_ section for more
+     discussion.
+
+   It is the view of the authors of this PEP that it's still
+   a good idea to use virtual environments even in
+   single-application container images. For instance, the
+   example from `Docker's Python image`_ would be better
+   implemented as::
+
+       FROM python:3
+       WORKDIR /usr/src/app
+       RUN python3 -m venv venv
+       RUN venv/bin/pip install -U pip
+       COPY requirements.txt ./
+       RUN venv/bin/pip install --no-cache-dir -r requirements.txt
+       COPY . .
+       CMD ["venv/bin/python", "./your-daemon-or-script.py"]
+
+   .. _`Docker's Python image`: https://hub.docker.com/_/python
+
+   This ensures that if you call out to OS-provided software
+   in your code, write a downstream ``Dockerfile`` that
+   applies OS upgrades, etc., any packages you've installed
+   cannot impact the OS.
+
+6. A Conda environment, where a non-``conda`` tool like pip
+   is being used to remove packages. In this context, Conda
+   acts as the external package manager and pip as the
+   Python-specific one. This is currently supported by Conda
+   as a way to install software not available in the Conda
+   repositories.
+
+   In some sense, this is similar to the first case, since
+   Conda provides its own installation of the Python
+   interpreter.
+
+   * Can you ``pip install`` without an override? Currently
+     yes; after this PEP yes, since Conda would not want to
+     add the marker to prevent pip from running.
+
+   * Can it delete externally-installed packages? Currently
+     yes; after this PEP yes, unless Conda changes their
+     ``sysconfig`` schemes to use separate ones for
+     pip-installed and Conda-installed software.  Currently
+     those are the same scheme, but it may be a good idea
+     for Conda to change these, because if pip removes a
+     package, it doesn't inform Conda of that.
+
+7. A build of Python as part of an integrated monorepo with
+   its own dependency tooling, such as a private monorepo at
```

Comment:
```suggestion
7. A build of Python as part of an integrated distribution with
   its own dependency tooling, such as a private base platform at
```

Reply:
See email, there's a different rephrasing which you may or may not prefer for this case.